### PR TITLE
Introduce wire scan MATLAB export functionality and improve data handling

### PR DESCRIPTION
**Pull request by kabanaty**
This pull request adds comprehensive unit tests for the MATLAB `.mat` export functionality and introduces a new convenience method to the `WireMeasurementAnalysisResult` class for exporting analysis results. The main focus is on ensuring the correctness and compatibility of `.mat` file exports, including detailed checks of the exported data structure and values.

**MATLAB Export Functionality and Testing:**

* Added a new method `to_mat` to the `WireMeasurementAnalysisResult` class, providing a simple interface to export results as MATLAB `.mat` files compatible with `wirescan_gui`.
* Introduced a new test module `tests/test_mat_export.py` that:
  - Provides synthetic test data generation for analysis results.
  - Includes unit tests for the `datetime_to_matlab_datenum` conversion utility.
  - Thoroughly validates the structure and content of `.mat` files produced by `analysis_result_to_mat`, checking field presence, data shapes, and value correctness.
  - Tests edge cases such as results with single profiles, missing BPMs, and method integration via the new `to_mat` method.

### CODE REVIEW

File: slac_measurements/wires/analysis/mat_export.py [hunk 1]
```diff
@@ -0,0 +1,646 @@
...
+        dt = dt.replace(tzinfo=timezone.utc)
+    return MATLAB_EPOCH_OFFSET + (dt - epoch).total_seconds() / 86400.0
+
+
+# ---------------------------------------------------------------------------
+# Name map (MAD → EPICS)
+# ---------------------------------------------------------------------------
+
+
+def _build_name_map(mad_names: list[str]) -> dict[str, str]:
```
Comment (eloise-nebula) [inline slac_measurements/wires/analysis/mat_export.py:49]: I think this function could be accomplished a lot more easily with something like:

``` python
import pykern.sql_db
import slac_db.device

def _build_name_map(mad_names: list[str]) -> dict[str, str]:
    try:
        name_map = {m: slac_db.device.get_attribute(m, 'cs_name') for m in mad_names}
    catch: pykern.sql_db.NoRows:
        # Handle device that isn't in db.
```

**Review by eloise-nebula — APPROVED** [review]
We discussed earlier this week that this is stopgap/temporary. If we're interested in doing Matlab translation again we should create a generalized technique.
